File: java/src/main/java/pl/kkurczewski/json/user/UserController.java
package pl.kkurczewski.json.user;

import pl.kkurczewski.json.user.dto.User;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {

    @PostMapping("/user")
    public User user(@RequestBody User user) {
        return user;
    }
}

File: kotlin/src/main/kotlin/pl/kkurczewski/json/user/UserController.kt
package pl.kkurczewski.json.user

import pl.kkurczewski.json.user.dto.User
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController {
    @PostMapping("/user")
    fun user(@RequestBody user: User) = user
}